### ISSUE TITLE
Issue with Elapsed, ETA and Remaining after upgrading OctoPrint-HomeAssistant Plugin

### ISSUE THREAD

**Issue by jegres1709** [e11en/ha-threedy-card#4]
Hello,

I´m facing a problem with invalid dates and wrong ETA via mqtt:
![image](https://github.com/e11en/ha-threedy-card/assets/101667335/272696ed-5591-44a8-ab51-1b181b19fd25)

reason is https://github.com/cmroche/OctoPrint-HomeAssistant/pull/108
Time will be shown as seconds. mreditor97 delivered a patch, but the formatted time is available in the attributes:
![image](https://github.com/e11en/ha-threedy-card/assets/101667335/b52c1631-d990-4a8c-813b-be4c9e53c213)
![image](https://github.com/e11en/ha-threedy-card/assets/101667335/87a3c585-fb9b-49e7-914b-6784017e295b)

is it possible to take the right time from the attributes or format it within the card?

Thank you!

**Comment by mreditor97** [comment]
I didn't know this card existed! Brilliant!

For the time being, it is possible to create yourself a template sensor, which converts the MQTT provided time (in seconds) to a timedelta, which allows for formatting in a pretty format. You could then use that template sensor within this card.

```yaml
template:
  - sensor:
      - name: Octoprint Formatted Time Left
        state: {{ timedelta(seconds=states('sensor.octoprint_time_left') | int(default=0)) }}
```

_This hasn't been tested yet_

I hope this helps!

**Comment by jegres1709** [comment]
thank you for the fast response and approach!

unfortunately it´s not possible to select the sensors in the card. it chooses the preselected sensors from the base_enttity
![image](https://github.com/e11en/ha-threedy-card/assets/101667335/a487af0a-ddec-4543-8ebb-23e11fa255de)

I suppose the card needs to be adjusted in this case

**Comment by mreditor97** [comment]
I see. The solution may require some updates to the card depending on how it works. We could maybe read the Formatted Time from one of the sensors instead of directly for example.

I will have a look more into what the card is doing, and see what solutions we have available to sort this for us.

**Comment by mreditor97** [comment]
@e11en I have been looking into a potential quality of life improvement, and I am thinking that adding in the ability to override the `base_entity` with a selectable entity for each sensor in the cases where people change the `entity_id` (like me who changed the Print Status to `sensor.octoprint_current_status` using the MQTT plugin). This would be a start for enabling more printer management programs like those mentioned in #2.

For this problem directly, we could do some detection to see if the received value is an `int`, and if it is then can be processed in seconds if not, it can be displayed how it currently.

What do you think of this solution?

**Comment by xZetsubou** [comment]
@mreditor97 actually I don't know if this repo is still active but I added some new features in my [fork](https://github.com/xZetsubou/ha-threedy-card) in case someone interesting
I haven't tested it yet but it also should fix #1 #2

**Comment by jegres1709** [comment]
thank you very much!
